registry/github/template.go

```diff
@@ -40,7 +40,7 @@ func (c *client) Template(u *library.User, s *registry.Source) ([]byte, error) {
 	//
 	// nolint: lll // ignore long line length due to variable names
 	data, _, resp, err := cli.Repositories.GetContents(context.Background(), s.Org, s.Repo, s.Name, opts)
-	if err != nil {
+	if err != nil && resp != nil {
```

Comment:
Do we think there is a potential for `err != nil` here but for `resp == nil`? 🤔 

If so, do we think there are any unintended consequences with this change?

I'm leaning towards not because in the above scenario, I'd still expect `data == nil`.

Which means we'd still return the `no Vela template found ...` error message.

Reply:
`GetContents` does return a pointer for `resp`. I didn't traverse all of the `GetContents` code paths but i believe it's possible. Regardless, we previously only handled the error when `resp` was assumed to be available so there's probably no harm in this change and we get a little extra protection.

But yea, to your point, i think we still get the same end effect: error message shown when `data` is `nil`.

Reply:
I think it's effectively the same, but it felt like something that should be checked if we're counting on it being there within the if statement.

I'm okay to remove the check if there's a drawback but if it's effectively the same, is it not better to err on the side of caution?

Reply:
I'm not proposing we remove the check to see if `resp != nil` 👍 

I think it makes sense to check if it's `nil`.

However, I brought it up in case there were any concerns but it doesn't sound like there is 😄 

My line of thought was if there are concerns with `err != nil` and `resp == nil`, we could try something like:

```go
data, _, resp, err := cli.Repositories.GetContents(context.Background(), s.Org, s.Repo, s.Name, opts)
if err != nil {
	if resp != nil && resp.StatusCode != http.StatusNotFound {
		return nil, fmt.Errorf("unexpected error fetching template %s/%s/%s: %v", s.Org, s.Repo, s.Name, err)
	}

	return nil, fmt.Errorf("no Vela template found at %s/%s/%s", s.Org, s.Repo, s.Name)
}
```

Reply:
I get what you're saying now. Thanks for spelling it out for me 😄  It *does* make sense to return different errors for this. Adding this tot he PR yet.